### ISSUE TITLE
Invalid multinomial distribution (sum of probabilities <= 0)

### ISSUE THREAD

**Issue by AlekzNet** [jcjohnson/torch-rnn#67]
After training a new 3 x1024 model (dropout=0.1, batchnorm=1) for a couple of days, I got the following error:

$  th mysample.lua -checkpoint /nas/doc/nn/checkpoint_200.t7.reset.t7 -length 2000 -gpu -1
Loading 	/nas/doc/nn/checkpoint_200.t7.reset.t7	
Loaded	
/home/alekz/torch/install/bin/luajit: bad argument #2 to '?' (invalid multinomial distribution (sum of probabilities <= 0) at /home/alekz/torch/pkg/torch/lib/TH/generic/THTensorRandom.c:120)
stack traceback:
	[C]: at 0x7ffad932e1d0
	[C]: in function 'multinomial'
	./LanguageModel.lua:196: in function 'sample'
	mysample.lua:43: in main chunk
	[C]: in function 'dofile'
	...lekz/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00405d70

What can be the problem?

**Comment by AlekzNet** [comment]
Apparently it's related to https://github.com/jcjohnson/torch-rnn/issues/65
